walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/util/controlflow/Result.kt

```diff
@@ -15,6 +16,31 @@ internal sealed class Result<out S> {
     class Failure(val payload: SdkException) : Result<Nothing>()
 }
 
+internal fun <T> KotlinResult<T>.toSDK(): Result<T> {
+    if (this.isSuccess) {
+        this.getOrNull()?.let {
+            return Result.Success<T>(it)
+        }
+    }
+    this.exceptionOrNull()?.let {
+        if (it is SdkException) {
+            return Result.Failure(it)
+        }
+    }
+    return Result.Failure(SdkException("Could not cast failure to SDK"))
```

Comment:
Could we append the message from the exception here so that if it is a failure from a dependency that is bubbled up, it would help in debugging the issue?